src/vcpkg/binarycaching.cpp

```diff
@@ -2168,29 +2156,119 @@ ExpectedS<std::vector<std::unique_ptr<IBinaryProvider>>> vcpkg::create_binary_pr
     return providers;
 }
 
-std::string vcpkg::reformat_version(const std::string& version, const std::string& abi_tag)
+namespace
 {
-    static const std::regex semver_matcher(R"(v?(\d+)(\.\d+|$)(\.\d+)?.*)");
+    struct ReformatParsedVersion
+    {
+        StringView major;
+        StringView minor;
+        StringView patch;
+    };
 
-    std::smatch sm;
-    if (std::regex_match(version.cbegin(), version.cend(), sm, semver_matcher))
+    StringView remove_leading_zeroes(StringView sv)
     {
-        auto major = trim_leading_zeroes(sm.str(1));
-        auto minor = sm.size() > 2 && !sm.str(2).empty() ? trim_leading_zeroes(sm.str(2).substr(1)) : "0";
-        auto patch = sm.size() > 3 && !sm.str(3).empty() ? trim_leading_zeroes(sm.str(3).substr(1)) : "0";
-        return Strings::concat(major, '.', minor, '.', patch, "-vcpkg", abi_tag);
+        if (sv.empty()) return "0";
+
+        auto it = std::find_if_not(sv.begin(), sv.end(), [](char ch) { return ch == '0'; });
+        if (it == sv.end())
+        {
+            // all zeroes - just return "0"
+            return StringView{sv.begin(), sv.begin() + 1};
+        }
+        else
+        {
+            return StringView{it, sv.end()};
+        }
     }
 
-    static const std::regex date_matcher(R"((\d\d\d\d)-(\d\d)-(\d\d).*)");
-    if (std::regex_match(version.cbegin(), version.cend(), sm, date_matcher))
+    // /(\d\d\d\d)-(\d\d)-(\d\d).*/
+    bool try_parse_reformat_date_version(ReformatParsedVersion& out, StringView version)
+    {
+        using P = vcpkg::Parse::ParserBase;
+        // a b c d - e f - g h <end>
+        // 0 1 2 3 4 5 6 7 8 9 10
+        if (version.size() < 10) return false;
+        auto first = version.begin();
+        if (!P::is_ascii_digit(*first++)) return false;
```

Comment:
Note to self/others: it would be really nice to not have yet another parser for this vs the one in DateVersion. No change requested.